lib/rouge/lexer.rb

```diff
@@ -198,6 +198,8 @@ def debug_enabled?
     protected
       # @private
       def register(name, lexer)
+        # reset an existing list of lexers
+        @all = nil if @all
```

Comment:
the `if` here is unnecessary, if it's nil it's nil.

Reply:
My intention was only to *reset* an existing `@all` instead of initializing an `@all` with a `nil` value. The `@all` initialization will be deferred until the call to method `:all`.

Reply:
@ashmaroli I have to admit, I'm somewhat new to benchmarking like this so I might have made a mistake but when I run this:

```ruby
require 'benchmark'

Benchmark.bm do |x|
  x.report { 100_000.times { @a = nil }}
  x.report { 100_000.times { @b = nil if @b }}
end
```

I get:

```
       user     system      total        real
   0.006432   0.000035   0.006467 (  0.006506)
   0.006502   0.000026   0.006528 (  0.006575)
```

Given this, does it make much of a difference if we perform the conditional check? If there's little to no additional overhead, it seems like setting `@all` to `nil` is simpler code.

(I really am new to doing this! Let me know if I've made a stupid mistake somewhere here.)

Reply:
@pyrmont Your benchmarks are not wrong. Just that our point-of-view ended up being different.
I chose to not initialize `@all = nil`  during calls to `:register`.

Consider the hypothetical situation where only the method `:register` was called 100 times without calling the `:all` method. At the end of all the calls, you'd have just the populated `@registry` ivar. (No trace of `@all`, i.e. *undefined*, or no memory allocated..)

Now, it is only when `:all` is called, the ivar `@all` gets defined and stored in memory.
Henceforth, any call to `:register` will reset the ivar `@all` to `nil` (will still occupy minimal memory) until the method `:all` is called to  get the current list of all lexers.

I can drop the check if you guys still feel that's better.

Reply:
No, that makes sense.